src/storage/lock_manager/util.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::Lock;
+use crate::storage::mvcc::Error as MvccError;
+use crate::storage::txn::Error as TxnError;
+use crate::storage::Error as StorageError;
+use crate::storage::Key;
+use crate::tikv_util::time;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+pub const PHYSICAL_SHIFT_BITS: usize = 18;
+
+// In milliseconds
+pub fn extract_physical_timestamp(ts: u64) -> u64 {
+    ts >> PHYSICAL_SHIFT_BITS
+}
+
+pub fn calculate_wait_for_lock_timeout(start_ts: u64, ttl: u64) -> u64 {
+    let limit = Duration::from_millis(extract_physical_timestamp(start_ts) + ttl);
+    let now = SystemTime::now()
```

Comment:
what if the current machine's logical time not correct or has a big difference with `pd`?

Reply:
It has been removed, and it doesn't matter.